From fb0a4029efbe1798c272b5b8fa0f0e8a3bbc58e2 Mon Sep 17 00:00:00 2001
From: Yue Tao <Yue.Tao@windriver.com>
Date: Mon, 22 Jan 2018 14:32:47 +0800
Subject: [PATCH 505/666] uio: fix section mismatch in fsl_srio_uio_probe

Section mismatch in reference from the function .fsl_srio_uio_probe() to the function .init.text:.srio_uio_setup(
The function .fsl_srio_uio_probe() references
the function __init .srio_uio_setup().
This is often because .fsl_srio_uio_probe lacks a __init
annotation or the annotation of .srio_uio_setup is wrong.

It turns out the srio_uio_setup is called outside __init section, hence we
need to remove the __init tag.

Signed-off-by: Yue Tao <Yue.Tao@windriver.com>
---
 drivers/uio/fsl_srio_uio.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/uio/fsl_srio_uio.c b/drivers/uio/fsl_srio_uio.c
index f015d33..5888ff1 100644
--- a/drivers/uio/fsl_srio_uio.c
+++ b/drivers/uio/fsl_srio_uio.c
@@ -125,7 +125,7 @@ static irqreturn_t srio_uio_irq_handler(int irq, struct uio_info *dev_info)
 	return IRQ_HANDLED;
 }
 
-static int __init srio_uio_setup(struct srio_dev *sriodev, u8 type, u32 port_id)
+static int srio_uio_setup(struct srio_dev *sriodev, u8 type, u32 port_id)
 {
 	int err;
 	struct srio_uio_info *info;
-- 
2.7.4

